exporter/azuredataexplorerexporter/factory_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuredataexplorerexporter
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/configtest"
+	"go.opentelemetry.io/collector/service/servicetest"
+)
+
+// Given a new factory and no-op exporter , the NewMetric exporter should work.
+// We could add additional failing tests if the config is wrong (using Validate) , but that is already done on config
+func TestCreateMetricsExporter(t *testing.T) {
+	factories, err := componenttest.NopFactories()
```

Comment:
In tests it's more fair to have copy-pasted sections, but still it seems like the setup here could be a common method. @yogilad WDYT?